pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -684,10 +689,24 @@ def _compile_objective_and_grad(self):
         )
         objective_grad = pt.grad(rewrite_pregrad(objective), budgets_flat)
 
-        objective_and_grad_func = function([budgets_flat], [objective, objective_grad])
+        if self.compile_kwargs and (self.compile_kwargs["mode"]).lower() == "jax":
+            # Use PyMC's JAX infrastructure for robust compilation
+            from pymc.sampling.jax import get_jaxified_graph
```

Comment:
Why did you need to do this instead of just compiling the same pytensor function with jax backend?